app/models/webhook/delivery.rb

```diff
@@ -0,0 +1,24 @@
+module Webhook
+  module Delivery
+    extend ActiveSupport::Concern
+
+    def generate_webhook_payload(action)
+      meta = {
+        model: self.class.name.underscore,
+        event: action,
+      }
+      webhook_renderer.render(self, root: :data, meta: meta)
+    end
+
+    def deliver_webhook(action)
```

Comment:
`generate_payload_and_send_webhook` would read easier and less ambiguous between deliver and send imho.